parse/parser.go

```diff
@@ -200,16 +202,44 @@ func makePrefixExpression(buf TokenBuffer) (ast.Expression, []error) {
 	return exp, errs
 }
 
-// TODO: implement me w/ test cases :-)
-// MakeInfixExpression retrieves infix parse function from map
-// then parse expression with that function if exist.
 func makeInfixExpression(buf TokenBuffer, exp ast.Expression, pre precedence) (ast.Expression, []error) {
-	return nil, nil
+	errs := make([]error, 0)
+	expression := exp
+	curTok := buf.Peek(CURRENT)
+
+	for pre < nextPrecedence(buf) {
+		fn := infixParseFnMap[buf.Peek(NEXT).Type]
+
```

Comment:
This is code style issue but I think this is quite eyesore to me.. Coding error handling right after definition is more readable:
```
fn := infixParseFnMap[buf.Peek(NEXT).Type]

if fn == nil {
```
to
```
fn := infixParseFnMap[buf.Peek(NEXT).Type]
if fn == nil {
```